### PR TITLE
Update doc about different configuration options

### PR DESCRIPTION
**Pull request by RassK**
### Background
Porting docs update from SignalFX (https://github.com/signalfx/signalfx-dotnet-tracing/pull/99). Modified for OTel.

### CODE REVIEW

File: docs/USAGE.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Use these environment variables to configure the tracing library:
 
 | Environment variable | Description | Default |
 |-|-|-|
-| `OTEL_TRACE_CONFIG_FILE` | The path to the configuration file. | `%CurrenctDIR%/datadog.json` |
+| `OTEL_TRACE_CONFIG_FILE` | Loads the json configuration file with configuration values. Overloaded value must be set via Environment variable. | ```{Environment.CurrentDirectory}/datadog.json``` |
```
Comment (pellared) [inline docs/USAGE.md:13]: How about `The file path of a JSON configuration file that will be loaded` ?

Comment (pellared) [inline docs/USAGE.md:13]: `{Environment.CurrentDirectory}/datadog.json` -> I suggest to use `CurrentWorkingDirectory` or just `WorkingDirectory` https://en.wikipedia.org/wiki/Working_directory

File: docs/USAGE.md [hunk 2]
```diff
@@ -62,6 +62,45 @@ Use these environment variables to configure the tracing library:
 | `OTEL_PROFILER_PROCESSES` | Sets the filename of executables the profiler can attach to. If not defined (default), the profiler will attach to any process. Supports multiple values separated with comma, for example: `MyApp.exe,dotnet.exe` |  |
 | `OTEL_PROFILER_EXCLUDE_PROCESSES` | Sets the filename of executables the profiler cannot attach to. If not defined (default), the profiler will attach to any process. Supports multiple values separated with comma, for example: `MyApp.exe,dotnet.exe` |  |
 
+## Ways to configure
+
+There are 3 ways to apply configuration values (Priority is from first to last):
+
+1. Environment Variables
+2. Web.config / App.config
+3. Json configuration file
```
Comment (pellared) [inline docs/USAGE.md:71]:
```suggestion
There are the following ways to apply configuration values (priority is from first to last):

1. [Environment variables)(#environment-variables)
2. [`web.config` or `app.config` file](#web.config-and-app.config)
3. [JSON configuration file](#json-configuration-file)
```

File: docs/USAGE.md [hunk 3]
```diff
@@ -62,6 +62,45 @@ Use these environment variables to configure the tracing library:
...
+
+### Environment variables
+
+Environment variables are the main way to configure values. After setting up environment variable, it can't be overridden by lower priority configuration option. 
+
+### Web.config and app.config
```
Comment (pellared) [inline docs/USAGE.md:77]:
```suggestion
### web.config and app.config
```

File: docs/USAGE.md [hunk 4]
```diff
@@ -62,6 +62,45 @@ Use these environment variables to configure the tracing library:
...
+
+When your application is running on .NET Framework (Does not work with .NET Core or later), all unset configuration values can be set via web configuration file (web.config) or application configuration file (app.config).
```
Comment (pellared) [inline docs/USAGE.md:79]:
```suggestion
For an application is running on .NET Framework, web configuration file (`web.config`) or application configuration file (`app.config`) can be used to configure settings.
```

File: docs/USAGE.md [hunk 5]
```diff
@@ -62,6 +62,45 @@ Use these environment variables to configure the tracing library:
...
+
+There are 3 ways to apply configuration values (Priority is from first to last):
+
+1. Environment Variables
+2. Web.config / App.config
+3. Json configuration file
+
+### Environment variables
+
+Environment variables are the main way to configure values. After setting up environment variable, it can't be overridden by lower priority configuration option.
```
Comment (pellared) [inline docs/USAGE.md:75]:
```suggestion
Environment variables are the main way to configure values.  A setting configured via an environment variable cannot be overridden.
```

File: docs/USAGE.md [hunk 6]
```diff
@@ -62,6 +62,45 @@ Use these environment variables to configure the tracing library:
...
+
+```xml
+<configuration>
+  <appSettings>
+    <add key="OTEL_SERVICE" value="my-service-name" />
+  </appSettings>
+</configuration>
+```
+
+### Json configuration file
```
Comment (pellared) [inline docs/USAGE.md:91]:
```suggestion
### JSON configuration file
```

File: docs/USAGE.md [hunk 7]
```diff
@@ -62,6 +62,45 @@ Use these environment variables to configure the tracing library:
...
+
+Json configuration file has the least priority. To use this option, all values that need to be overridden must be unset at first. By default if `OTEL_TRACE_CONFIG_FILE` is unset, the application is searching for `datadog.json` in current working directory (acquired by `Environment.CurrentDirectory`).
```
Comment (pellared) [inline docs/USAGE.md:93]:
```suggestion
By default, if `OTEL_TRACE_CONFIG_FILE` is unset, the application is searching for `datadog.json` in the current working directory (acquired by [`Environment.CurrentDirectory`](https://docs.microsoft.com/en-us/dotnet/api/system.environment.currentdirectory?view=net-5.0)).
```

**Review by pellared — APPROVED** [review]

File: docs/USAGE.md [hunk 8]
```diff
@@ -10,7 +10,7 @@ Use these environment variables to configure the tracing library:
 
 | Environment variable | Description | Default |
 |-|-|-|
-| `OTEL_TRACE_CONFIG_FILE` | The path to the configuration file. | `%CurrenctDIR%/datadog.json` |
+| `OTEL_TRACE_CONFIG_FILE` | The file path of a JSON configuration file that will be loaded. | ```%WorkingDirectory%/datadog.json``` |
```
Comment (pjanotti) [inline docs/USAGE.md:13]: @dszmigielski this name here needs to be also updated.

**Review by pjanotti — APPROVED** [review]
LGTM

**Review by nrcventura — APPROVED** [review]

**Pull request merged**